src/components/playback/playbackmanager.js

```diff
@@ -261,16 +261,59 @@ define(['events', 'datetime', 'appSettings', 'itemHelper', 'pluginManager', 'pla
 
     function getIntros(firstItem, apiClient, options) {
 
-        if (options.startPositionTicks || options.startIndex || options.fullscreen === false || !enableIntros(firstItem) || !userSettings.enableCinemaMode()) {
+        var unwatchedOnly = userSettings.enableCinemaTrailersUnseenOnly();
+        var trailerCount = userSettings.cinemaModeTrailerCount();
+        var enableCinema = userSettings.enableCinemaMode();
+
+        if (options.startPositionTicks || options.startIndex || options.fullscreen === false || !enableIntros(firstItem) || !enableCinema || trailerCount == 0) {
             return Promise.resolve({
                 Items: []
             });
         }
 
-        return apiClient.getIntros(firstItem.Id).then(function (result) {
+        var fetchOptions = {
+            limit: 500,
+            fields: "RemoteTrailers,LocalTrailerCount,ServerID",
+            filters: unwatchedOnly ? "IsUnPlayed" : "",
+            recursive: true,
+            includeItemTypes: "Movie"
+        };
+
+        function getAllTrailers(item) {
+            if (item.LocalTrailerCount) {
+                return apiClient.getLocalTrailers(apiClient.getCurrentUserId(), item.Id);
+            } else {
+                var remoteTrailers = item.RemoteTrailers || [];
 
-            return result;
+                if (!remoteTrailers.length) {
+                    return Promise.reject();
+                }
+
+                return remoteTrailers.map(function (t) {
+                    return {
+                        Name: t.Name || (item.Name + ' Trailer'),
+                        Url: t.Url,
+                        MediaType: 'Video',
+                        Type: 'Trailer',
+                        ServerId: apiClient.serverId()
+                    };
+                });
+            }
+        }
+
+        return apiClient.getItems(apiClient.getCurrentUserId(), fetchOptions).then(function (unwatchedMoves) {
+            var randomTrailers = [];
+
+            for (var i=0; i < trailerCount; i++) {
+                var index = Math.floor(Math.random() * unwatchedMoves.Items.length);
+                var randomMovie = unwatchedMoves.Items[index];
+                var trailer = getAllTrailers(randomMovie)[0];
```

Comment:
`apiClient.getLocalTrailers` returns `Promise`, so I think that `getAllTrailers(randomMovie)[0]` won't work properly.
I cannot cleanly check this because I cannot get local trailers to be detected - just forced `apiClient.getLocalTrailers` call.

I think you should split local and remote trailers
* get unwatched moves (_done already_)
* randomly choose movies with remote trailers and/or local trailers
* generate array of `apiClient.getLocalTrailers` for local trailers
* `Promise.all` local trailers, combine result with remote ones
* randomly construct `{ Items }` from combined result

Reply:
I don't have any local trailers myself so I couldn't test it.
I made all you suggested changes and I think it should work now.